vignettes/fims-setup-guide.Rmd

```diff
@@ -0,0 +1,177 @@
+    header_line <- lines[start_idx]
+    header_level <- nchar(gsub("(^#+).*", "\\1", header_line))
+    
+    # Find headers that are at the SAME level or HIGHER (fewer or equal #)
+    # We want to ignore ### if we are looking for a ## section
+    stop_pattern <- paste0("^#{1,", header_level, "} ")
+    all_potential_stops <- grep(stop_pattern, lines)
+    
+    # The end index is the next header that isn't the current one
+    end_idx <- all_potential_stops[all_potential_stops > start_idx][1]
+    
+    if (is.na(end_idx)) end_idx <- length(lines) else end_idx <- end_idx - 1
+    
+    # Extract and add blank lines for proper RMarkdown list rendering
+    content <- c("", lines[(start_idx + 1):end_idx], "") 
+    writeLines(content, tmp)
+  } else {
+    writeLines("_Section not found in remote documentation._", tmp)
+  }
+  
+  return(tmp)
+}
+```
+
+## Overview
+
+This guide facilitates the configuration of a standardized Fisheries Integrated Modeling System (FIMS) environment. By utilizing cloud-based workstations and containerized development environments, we ensure cross-platform consistency and eliminate the complexities of manual dependency management.
+
+Please follow the section below that matches how you plan to set up FIMS, i.e., ☁️[Google Cloud Workstations](#google-cloud-workstations) for managed cloud IDEs (internal NOAA), 🌐[GitHub Codespaces](#github-codespaces) for browser-based containerized environments, 🐧[Windows Subsystem for Linux 2 (WSL2)](#wsl2) for a containerized Linux environment running on Windows hardware, and 🧑‍💻[Local](#local) installation for direct deployment on your machine's primary operating system.
+
+```{r common-setup, echo=FALSE}
+# The bash script for setting up FIMS
+common_script <- "bash <(curl -sL https://raw.githubusercontent.com/NOAA-FIMS/FIMS/main/setup_fims.sh)"
+
+# Fetch the remote R script for the data bucket example
+readwrite_databucket_url <- "https://raw.githubusercontent.com/nmfs-opensci/CloudComputingSetup/refs/heads/main/R/readwrite_databucket.R"
+readwrite_databucket_script <- tryCatch(
+  readLines(readwrite_databucket_url, warn = FALSE),
+  error = function(e) "# Could not load remote script."
+)
+
+# Fetch the remote terminal script for mounting a data bucket
+mount_bucket_url <- "https://raw.githubusercontent.com/nmfs-opensci/CloudComputingSetup/refs/heads/main/R/mount_bucket.R"
+mount_bucket_script <- tryCatch(
+  readLines(mount_bucket_url, warn = FALSE),
+  error = function(e) "# Could not load remote script."
+)
+```
+
+## ☁️ Google Cloud Workstations
+
+NOAA Fisheries Google Cloud Workstations provide a managed RStudio, Visual Studio Code (VS Code), or Positron environment. For detailed visual guides (e.g., snapshots) on starting, stopping, or managing workstations, please refer to the [Internal NOAA documentation](https://docs.google.com/document/d/1nziPdPULoRWOYQ9WKzISNUgJvANACvfYpFr1z3Ro2Bc/edit?tab=t.0#heading=h.5509095bxzv7).
```

Comment:
Consider changing Visual Studio Code to "Code OSS (the open source version of VS Code)"